### ISSUE TITLE
DEL with wrong gene annotation

### ISSUE THREAD

**Issue by alsanju** [broadinstitute/gatk-sv#104]
### Affected module(s) or script(s)
Module 08 - Annotation

### Description and actual behavior
The following deletion overlaps two protein coding genes (PRKD1 and FOXG1), but only PRKD1  is being annotated as PROTEIN_CODING__LOF, while FOXG1 is annotated as PROTEIN_CODING__INTRONIC. The post-annotation entry in the VCF is:

```
CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
chr14	27766767	.	N	<DEL>	1	PASS	END=30144669;SVTYPE=DEL;CHR2=chr14;SVLEN=2377902;ALGORITHMS=depth,manta;EVIDENCE=BAF,PE,RD,SR;PROTEIN_CODING__LOF=PRKD1;PROTEIN_CODING__INTRONIC=FOXG1;LINCRNA__INTRONIC=AL133372.2,AL138974.1,AL139023.1,AL158058.1,LINC01551,LINC02281,LINC02282,LINC02300,LINC02326,LINC02327;NONCODING_SPAN=DNase,Enhancer,HAR,Tommerup_TADanno;NONCODING_BREAKPOINT=Tommerup_TADanno;AN=1428;AC=1;AF=0.0007;N_BI_GENOS=714;N_HOMREF=713;N_HET=1;N_HOMALT=0;FREQ_HOMREF=0.998599;FREQ_HET=0.00140056;FREQ_HOMALT=0;MALE_AN=772;MALE_AC=0;MALE_AF=0;MALE_N_BI_GENOS=386;MALE_N_HOMREF=386;MALE_N_HET=0;MALE_N_HOMALT=0;MALE_FREQ_HOMREF=1;MALE_FREQ_HET=0;MALE_FREQ_HOMALT=0;FEMALE_AN=656;FEMALE_AC=1;FEMALE_AF=0.001524;FEMALE_N_BI_GENOS=328;FEMALE_N_HOMREF=327;FEMALE_N_HET=1;FEMALE_N_HOMALT=0;FEMALE_FREQ_HOMREF=0.996951;FEMALE_FREQ_HET=0.00304878;FEMALE_FREQ_HOMALT=0
```

Entries for the gene that should’ve been annotated in the `AnnotateVcf.protein_coding_gtf` file used:

```
chr14	HAVANA	gene	28760330	28770277	.	+	.	gene_id "ENSG00000176165.10"; gene_type "protein_coding"; gene_name "FOXG1"; level 2; havana_gene "OTTHUMG00000140187.6";
chr14	HAVANA	gene	29576479	30191898	.	-	.	gene_id "ENSG00000184304.14"; gene_type "protein_coding"; gene_name "PRKD1"; level 2; tag "overlapping_locus"; havana_gene "OTTHUMG00000140203.5"
````

#### Expected behavior
FOXG1 gene should also be annotated as PROTEIN_CODING__LOF.

**Comment by epiercehoffman** [comment]
Apologies for the delay following up on this.

I compared the entries for both genes in the `protein_coding_gtf` file and found that PRKD1 has a `CDS` and an `exon` entry for each exon, while FOXG1 only has an `exon` entry for each exon (see snippets below). It turns out our annotation scripts only use the `CDS` entries as evidence for exonic overlap, rather than `exon` or `gene`. I believe that this behavior is correct, as it appears that the `exon` label may be used for UTR regions in the GTF as well, whereas `CDS` exclusively refers to coding sequence regions. You can see this in the snippets below, as the `exon` regions of FOXG1 are tagged `"alternative_5_UTR"`; they also have a lower level of transcript support (5 is worse than 3 for PRKD1). However, I cannot speak to whether the GTF file itself is correct/complete for FOXG1, as it also says FOXG1 is `protein_coding` yet provides no actual `CDS` regions. I will follow up about this, as well as confirming this is the desired behavior of the annotation scripts.

FOXG1 GTF entries:

```
chr14	HAVANA	exon	28760330	28760359	.	+	.	gene_id "ENSG00000176165.10"; transcript_id "ENST00000636604.1"; gene_type "protein_coding"; gene_name "FOXG1"; transcript_type "processed_transcript"; transcript_name "FOXG1-204"; exon_number 1; exon_id "ENSE00003792225.1"; level 2; transcript_support_level "5"; tag "alternative_5_UTR"; tag "RNA_Seq_supported_only"; havana_gene "OTTHUMG00000140187.6"; havana_transcript "OTTHUMT00000490688.1";
chr14	HAVANA	gene	28760330	28770277	.	+	.	gene_id "ENSG00000176165.10"; gene_type "protein_coding"; gene_name "FOXG1"; level 2; havana_gene "OTTHUMG00000140187.6";
chr14	HAVANA	transcript	28760330	28765747	.	+	.	gene_id "ENSG00000176165.10"; transcript_id "ENST00000636604.1"; gene_type "protein_coding"; gene_name "FOXG1"; transcript_type "processed_transcript"; transcript_name "FOXG1-204"; level 2; transcript_support_level "5"; tag "alternative_5_UTR"; tag "RNA_Seq_supported_only"; havana_gene "OTTHUMG00000140187.6"; havana_transcript "OTTHUMT00000490688.1";
chr14	HAVANA	exon	28765678	28765747	.	+	.	gene_id "ENSG00000176165.10"; transcript_id "ENST00000636604.1"; gene_type "protein_coding"; gene_name "FOXG1"; transcript_type "processed_transcript"; transcript_name "FOXG1-204"; exon_number 2; exon_id "ENSE00003793213.1"; level 2; transcript_support_level "5"; tag "alternative_5_UTR"; tag "RNA_Seq_supported_only"; havana_gene "OTTHUMG00000140187.6"; havana_transcript "OTTHUMT00000490688.1";
```

PRKD1 GTF entries:

```
chr14	HAVANA	gene	29576479	30191898	.	-	.	gene_id "ENSG00000184304.14"; gene_type "protein_coding"; gene_name "PRKD1"; level 2; tag "overlapping_locus"; havana_gene "OTTHUMG00000140203.5";
chr14	HAVANA	CDS	29663812	29663859	.	-	2	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 6; exon_id "ENSE00002426711.1"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	exon	29663812	29663859	.	-	.	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 6; exon_id "ENSE00002426711.1"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	transcript	29663812	30191898	.	-	.	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	CDS	29666077	29666208	.	-	2	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 5; exon_id "ENSE00001759247.1"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	exon	29666077	29666208	.	-	.	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 5; exon_id "ENSE00001759247.1"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	CDS	29725536	29725674	.	-	0	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 4; exon_id "ENSE00001321203.2"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	exon	29725536	29725674	.	-	.	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 4; exon_id "ENSE00001321203.2"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	CDS	30047710	30047742	.	-	0	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 3; exon_id "ENSE00002353662.1"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	exon	30047710	30047742	.	-	.	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 3; exon_id "ENSE00002353662.1"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	start_codon	30047740	30047742	.	-	0	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 3; exon_id "ENSE00002353662.1"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	UTR	30047743	30047787	.	-	.	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 3; exon_id "ENSE00002353662.1"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	UTR	30191299	30191450	.	-	.	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 2; exon_id "ENSE00002402086.1"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";
chr14	HAVANA	UTR	30191866	30191898	.	-	.	gene_id "ENSG00000184304.14"; transcript_id "ENST00000549503.1"; gene_type "protein_coding"; gene_name "PRKD1"; transcript_type "protein_coding"; transcript_name "PRKD1-207"; exon_number 1; exon_id "ENSE00002427432.1"; level 2; protein_id "ENSP00000446866.1"; transcript_support_level "3"; tag "downstream_ATG"; tag "mRNA_end_NF"; tag "cds_end_NF"; havana_gene "OTTHUMG00000140203.5"; havana_transcript "OTTHUMT00000409377.2";

```

**Comment by mwalker174** [comment]
Fixed in #342
